src/Controls/DateTime/DateTimeParser.php

```diff
@@ -0,0 +1,86 @@
+<?php declare(strict_types = 1);
+
+namespace Contributte\Forms\Controls\DateTime;
+
+use DateTimeImmutable;
+use DateTimeInterface;
+use DateTimeZone;
+use Nette\SmartObject;
+use function is_string;
+use function strlen;
+
+class DateTimeParser
+{
+
+	use SmartObject;
+
+	/** @var string[] */
+	private $formats = [];
+
+	/** @var ?DateTimeZone */
+	private $defaultTimeZone = null;
+
+	/** @var string[] */
+	private $defaultFormats = [
```

Comment:
Docs says [date/time value is always formatted YYYY-MM-DDThh:mm](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local). Why do we need to test datetime string against so many formats?

Reply:
Oh, I noticed you also support a text input. Now it makes sense!

Reply:
Exactly. Also it is common se use some common format obtained from database/API/session/whatever in setValue/setDefaultValue

Reply:
And we have also input type "date" that returns yyyy-mm-dd

Reply:
It makes sense to me using it for text input. Setting "random" format datetime string in PHP is not something i prefer. I always try to work with objects datetime or brick/*.

Reply:
I also preffer to work with datetime objects whenever possible but it is common use-case to initialize form by array/object returned from some persistence layer that returns it as string.

Reply:
And it seems like popular feature because existing implementations (nextras, contributte/forms-bootstrap, ..) have it in some way.

In our codebase we need it in one place I can recall because we use datagrid that serializes current filters to query string and re-inits filter inputs by it.